rotkehlchen/assets/resolver.py

```diff
@@ -25,20 +27,30 @@
 
 class AssetResolver:
     __instance: Optional['AssetResolver'] = None
+    _globaldb: 'GlobalDBHandler'
+    _constant_assets: set['Asset']
     # A cache so that the DB is not hit every time
     # the cache maps identifier -> final representation of the asset
     assets_cache: LRUCacheLowerKey['AssetWithNameAndType'] = LRUCacheLowerKey(maxsize=512)
     types_cache: LRUCacheLowerKey[AssetType] = LRUCacheLowerKey(maxsize=512)
 
-    def __new__(cls) -> 'AssetResolver':
+    def __new__(
+            cls,
+            globaldb: 'GlobalDBHandler | None' = None,
+            constant_assets: set['Asset'] | None = None,
+    ) -> 'AssetResolver':
         """Lazily initializes AssetResolver
 
         It always uses the GlobalDB to resolve assets
         """
         if AssetResolver.__instance is not None:
             return AssetResolver.__instance
 
+        assert globaldb is not None
+        assert constant_assets is not None
         AssetResolver.__instance = object.__new__(cls)
+        AssetResolver._globaldb = globaldb
+        AssetResolver._constant_assets = constant_assets
```

Comment:
this is duplicating constant assets in memory right? Shouldn't we somehow make sure the set is only kept here as it's kept here only?

Reply:
it's not as it's a mutable type and as such is just passed by reference